File: src/main/java/cz/wa/secretagent/view/renderer/model/WeaponModelDrawer.java
//package cz.wa.secretagent.view.renderer.model;
//
//import java.util.List;
//
//import org.apache.commons.math3.geometry.euclidean.twod.Vector2D;
//import org.springframework.beans.factory.annotation.Required;
//
//import cz.wa.secretagent.view.texture.DrawBounds;
//import cz.wa.secretagent.view.texture.TextureToDraw;
//import cz.wa.secretagent.world.entity.Entity;
//import cz.wa.secretagent.worldinfo.WorldHolder;
//import secretAgent.view.model.AnimatedModel;
//import secretAgent.view.model.WeaponModel;
//import secretAgent.view.renderer.model.AbstractModelDrawer;
//
///**
// * Draws weapon model, used when drawing on screen only!
// * Weapon in world is drawn with the agent.
// *
// * @author Ondrej Milenovsky
// */
//public class WeaponModelDrawer extends AbstractModelDrawer<WeaponModel> {
//
//    private static final long serialVersionUID = 4014057635662109586L;
//
//    private WorldHolder worldHolder;
//
//    @Override
//    public void draw(WeaponModel model, Entity entity, Vector2D pos, double scale) {
//        AnimatedModel model2 = model.getModel(false);
//        List<TextureToDraw> textures = model2.getTextures();
//        TextureToDraw tex = getFrame(getWorldHolder().getWorld().getLevelMap().getTimeMs(),
//                model2.getDurationMs(), textures);
//        primitivesDrawer.drawTexture(tex, pos, new DrawBounds(model.getBounds()), scale);
//    }
//
//    public WorldHolder getWorldHolder() {
//        return worldHolder;
//    }
//
//    @Required
//    public void setWorldHolder(WorldHolder worldHolder) {
//        this.worldHolder = worldHolder;
//    }
//
//}

File: src/main/kotlin/secretAgent/view/renderer/model/WeaponModelDrawer.kt
package secretAgent.view.renderer.model

import cz.wa.secretagent.worldinfo.WorldHolder
import org.apache.commons.math3.geometry.euclidean.twod.Vector2D
import secretAgent.view.model.WeaponModel
import secretAgent.view.renderer.DrawBounds
import secretAgent.world.entity.Entity

/**
 * Draws weapon model, used when drawing on screen only!
 * Weapon in world is drawn with the agent.
 *
 * @author Ondrej Milenovsky
 */
class WeaponModelDrawer : AbstractModelDrawer<WeaponModel>() {

    lateinit var worldHolder: WorldHolder

    override fun draw(model: WeaponModel, entity: Entity?, pos: Vector2D, scale: Double) {
        val model2 = model.getModel(false)
        val textures = model2.textures
        val tex = AbstractModelDrawer.getFrame(worldHolder.world.levelMap.timeMs, model2.durationMs, textures!!)
        primitivesDrawer.drawTexture(tex!!, pos, DrawBounds(model.bounds), scale)
    }

    companion object {
        private const val serialVersionUID = 4014057635662109586L
    }
}
